.docker/entrypoint.sh

```
@@ -14,21 +14,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Exit immediately with failure status if any command fails
-set -e
 cd source/databricks/tests/
 
 # There env vars are important to ensure that the driver and worker nodes in spark are alligned
 export PYSPARK_PYTHON=/opt/conda/bin/python
 export PYSPARK_DRIVER_PYTHON=/opt/conda/bin/python
 
+# Writing output to log with 'tee' cause exit code to be '0' even if tests fails
+coverage run --branch -m pytest -n 2 . | tee pytest-results.log
+# If test summary contains errors we return exit code 2 to signal we want to retry
+matchTestErrors=$(grep -Po '^=+.* [[:digit:]]+ errors.* in .*=+$' pytest-results.log)
```

There is a risk that if the output in the textfile changes, i.e. with new updates to the test framework or similar, this check will fail. Can we do something, so that we also fail, if the expected "0 errors found" is not present in the log file? Otherwise we might end up in unexpected false positives or false negatives as a sideeffect to a test framework update

---

Changes to the output is a risk, so in that sense this technique is brittle. We cannot do what you sugest, since the test summary (output) changes based on what fails. But I will add an example output for each regex so other better understand what the regex finds.